pyproject.toml

```diff
@@ -16,14 +16,17 @@ python = "^3.8"
 requests = "^2"
 requests_oauthlib = "^1"
 msal = "^1"
+sortedcontainers = "^2.3"
```

Comment:
What is this being used for btw?

Reply:
Used in datapoints fetching for two reasons:
- Keep a sorted list of fetch tasks so that when we subdivide the time domain an unknown number of times, the order is automatically kept (for limited queries, where we repeatedly check if earlier tasks have fetched enough points (self-kill- and future-tasks-kill measure), the time savings of not having to sort-on-demand is huge)
- Use a storage container for datapoints that keeps the correct order to make final array or list creation O(n) - and for limited queries - to make it super easy (indexing-wise) to cut-off at limit)

Reply:
I read through the release notes and I think we could lower the requirement from 2.3 to 2.2 which fixed a bug with CPython ref counting